### PR TITLE
Improve github ci builds

### PR DESCRIPTION
**Pull request by ajaygeorge**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -319,7 +478,7 @@ jobs:
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
           ./mvnw install ${MAVEN_FAST_INSTALL} -am -pl $(echo '${{ matrix.modules }}' | cut -d' ' -f1)
       - name: Maven Tests
-        run: ./mvnw test ${MAVEN_TEST} -pl ${{ matrix.modules }}
+        run: $RETRY ./mvnw test ${MAVEN_TEST} -pl ${{ matrix.modules }}
```
Comment (aweisberg) [inline .github/workflows/ci.yml:481]: I don't think we should retry the tests? That is just hiding flaky tests.

**Comment by aweisberg** [comment]
Sorry I didn't get to this. I got busy for quite some time. If you want to rebase we can see if it runs and then I will review?

**Comment by ajaygeorge** [comment]
> Sorry I didn't get to this. I got busy for quite some time. If you want to rebase we can see if it runs and then I will review?

Let me rebase and get some stats. The last time I tried running it to get some clean stats I had to find a window where there was no other builds running.

Regarding retry for tests, I agree about the hiding of flaky tests part. What all do you think we should retry?
Also, shall we fast fail if `ci / maven-checks` (the basic check) fails. Right now I see if there is a checkstyle issue we still end up running all the steps which seems wasteful.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ajaygeorge) [reply .github/workflows/ci.yml]: removed it.

**Pull request closed without merging**